packages/python-packages/doc-warden/warden/WardenConfiguration.py

```diff
@@ -74,6 +74,18 @@ def __init__(self):
             dest = 'root_check_enabled',
             required = False,
             help = 'Enable or disable checking for a readme at the root of the repository. Defaults true. Overrides .docsettings contents.')
+        parser.add_argument(
+            '-s',
+            '--pipeline-stage',
+            dest = 'pipeline_stage',
+            required = False,
+            help = 'Specify the stage of the pipeline. Used to provide conditional functionality depending on the stage of the pipeline')
+        parser.add_argument(
+            '-t',
+            '--target-file',
+            dest = 'target_file',
```

Comment:
I am not a fan of this free-form argument here. The pattern for a `warden` run lifespan is.

1. start-up, figure out configuration
2. leverage configuration to execute tasks (`index`, `validate readme content`, `validate changelog content`, `validate readme presence`)

The lifespan you're introducing here is to _change the configuration_ as `target_file` needs to set to a specific value to run `validate_changelog` or `validate_readme`. 

It _isn't_ a huge deal that we're updating the `config.target_files` attribute, but I honestly don't even want to think about 

> what state is the configuration is at this point?

It should be a reliable dictionary only.